### DISCUSSION TITLE
Readd OnSelfRagdoll

### DISCUSSION THREAD

**Discussion by SquidingTin** [FakeFishGames/Barotrauma#12959] [Ideas and suggestions]
This was removed in the 1.1.18.0 update for unknown reasons, this status type didnt have any issues, in fact it worked just fine. i know the devs no longer used it but that doesnt mean it has to be removed? Modders clearly used it as im here complaining about it

**Comment by Regalis11** [comment]
This was removed because 
1. It had never worked properly. The way the effect was executed very inconsistently between the clients and servers, which lead to various syncing issues when you e.g. used the old Dual Spec-Ops talent.
2. It was no longer used in the vanilla game, and we felt it was better to just remove it than spend time getting it to work.

I think we didn't expect many mods to be using it though, I'm sorry that we ended up removing functionality that your relied on. :/ I wonder if there's another way to do what you're doing though, could the `OnRagdoll` effect type be used instead for example?

**Reply by SquidingTin** [reply]
Do not worry, i think theres a way to do this with onragdoll, mainly checking if the player has stun, as when your knocked out or stunned, you have that affliction but not when you hold space
